cmd/operator-sdk/generate/csv.go

```diff
@@ -97,60 +106,42 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
 
 func (c csvCmd) run() error {
 
-	absProjectPath := projutil.MustGetwd()
-	cfg := &input.Config{
-		AbsProjectPath: absProjectPath,
-		ProjectName:    filepath.Base(absProjectPath),
-	}
-	if projutil.IsOperatorGo() {
-		cfg.Repo = projutil.GetGoPkg()
-	}
-
 	log.Infof("Generating CSV manifest version %s", c.csvVersion)
 
-	csvCfg, err := catalog.GetCSVConfig(c.csvConfigPath)
-	if err != nil {
-		return err
-	}
 	if c.operatorName == "" {
-		// Use config operator name if not set by CLI, i.e. prefer CLI value over
-		// config value.
-		if c.operatorName = csvCfg.OperatorName; c.operatorName == "" {
-			// Default to using project name if both are empty.
-			c.operatorName = filepath.Base(absProjectPath)
-		}
-	}
-
-	s := &scaffold.Scaffold{}
-	csv := &catalog.CSV{
-		CSVVersion:     c.csvVersion,
-		FromVersion:    c.fromVersion,
-		ConfigFilePath: c.csvConfigPath,
-		OperatorName:   c.operatorName,
+		c.operatorName = filepath.Base(projutil.MustGetwd())
 	}
-	err = s.Execute(cfg, csv)
-	if err != nil {
-		return fmt.Errorf("catalog scaffold failed: %v", err)
+	cfg := gen.Config{
+		OperatorName: c.operatorName,
+		Inputs:       c.inputs,
+		OutputDir:    c.outputDir,
 	}
 
-	gcfg := gen.Config{
-		OperatorName: c.operatorName,
-		OutputDir:    filepath.Join(gencatalog.OLMCatalogDir, c.operatorName),
+	csv := gencatalog.NewCSV(cfg, c.csvVersion, c.fromVersion)
+	if err := csv.Generate(); err != nil {
+		return fmt.Errorf("error generating CSV: %v", err)
 	}
-	pkg := gencatalog.NewPackageManifest(gcfg, c.csvVersion, c.csvChannel, c.defaultChannel)
+	pkg := gencatalog.NewPackageManifest(cfg, c.csvVersion, c.csvChannel, c.defaultChannel)
 	if err := pkg.Generate(); err != nil {
 		return fmt.Errorf("error generating package manifest: %v", err)
 	}
 
 	// Write CRD's to the new or updated CSV package dir.
 	if c.updateCRDs {
-		input, err := csv.GetInput()
+		// TODO: Add a "crd" key to the --inputs flag to allow
+		// configuring the location of the CRD manifests directory
+		// from which to find and update CRD's.
+		crdManifestSet, err := findCRDFileSet(c.inputs[gencatalog.DeployDirKey])
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to update CRD's: %v", err)
 		}
-		err = writeCRDsToDir(csvCfg.CRDCRPaths, filepath.Dir(input.Path))
-		if err != nil {
-			return err
+		// TODO: This path should come from the CSV generator field csvOutputDir
+		bundleDir := filepath.Join(c.outputDir, gencatalog.OLMCatalogChildDir, c.operatorName, c.csvVersion)
+		for path, b := range crdManifestSet {
+			path = filepath.Join(bundleDir, path)
+			if err = ioutil.WriteFile(path, b, fileutil.DefaultFileMode); err != nil {
+				return fmt.Errorf("failed to update CRD's: %v", err)
```

Comment:
Is not the command just creating/updating CSV manifests? So, I think that this msg can bring a wrong understanding.

Reply:
No this part of the code is after we've generated the CSV and Package Manifest files.
This part is purely to take the existing CRDs in `deploy/crds/...` and update them in the bundle directory where we generated the CSV based on the `--update-crd` flag.
So it's actually creating/updating CRDs.